grant/evalutation/license_evalutation.go

```diff
@@ -44,19 +44,6 @@ func checkSBOM(ec EvaluationConfig, sb sbom.SBOM) LicenseEvaluations {
 }
 
 func checkLicense(ec EvaluationConfig, pkg *grant.Package, l grant.License) LicenseEvaluation {
-	if !l.IsSPDX() && ec.CheckNonSPDX {
-		if denied, rule := ec.Policy.IsDenied(l, pkg); denied {
-			var reason Reason
-			if rule != nil {
-				reason = Reason{
-					Detail:   ReasonLicenseDeniedPolicy,
-					RuleName: rule.Name,
-				}
-			}
-			return NewLicenseEvaluation(l, pkg, ec.Policy, []Reason{reason}, false)
-		}
-	}
```

Comment:
If I'm not mistaken, the behavior was the same no matter whether it's SPDX or not here: https://github.com/anchore/grant/blob/082dc4846338931734b7090e45934e63f8d130a3/grant/evalutation/license_evalutation.go#L47-L77

I believe it `ec.CheckNonSPDX` should be passed to `ec.Policy.IsDenied(l, pkg)` but I do not know what is the expected behavior. If indeed the code previous code was invalid, then it could be handled in another PR.

Reply:
I can take this on and move `ec.CheckNonSPDX` into the `ec.Policy.IsDenied`

The previous code was invalid.